### DISCUSSION TITLE
Allow grafana to receive alerts from other sources

### DISCUSSION THREAD

**Discussion by damienleger** [grafana/grafana#45773] [Alerting]
Hello!

**What happened**:

I'm running prometheus-operator on k8s and Grafana 8.1.3

I have a bunch of alerts set in prometheus-operator through ServiceMonitor or Probe (for blackbox-exporter) k8s resources. I see them in the alert panel [1] and one is firing.

I have setup notification system to send POST request to https://webhook.site/ because it's convenient for testing [2] [3] [4]. But no notifications are sent at all I don't know why, and debug logs aren't helping [5].

[1]
![image](https://user-images.githubusercontent.com/16499022/133401632-1b81208f-fa7d-4483-adc6-be735cd49552.png)

[2]
![image](https://user-images.githubusercontent.com/16499022/133401732-e387e9b8-2707-47d9-9c9d-ffd4610161b7.png)

[3]
![image](https://user-images.githubusercontent.com/16499022/133401795-6a4ad4cc-ed3d-46cc-baf2-dd674f170f60.png)

[4]
![image](https://user-images.githubusercontent.com/16499022/133401851-519ed1b5-15a5-4ea9-831f-29b00a35bf4d.png)

[5]
![image](https://user-images.githubusercontent.com/16499022/133403351-733f3273-bfa2-4ae5-bfd1-14085ff706a9.png)

**What you expected to happen**:

To see a notification every 10s.

**How to reproduce it (as minimally and precisely as possible)**:

Have some alert firing set in prometheus. Go to https://webhook.site/ and get the URL from there to configure the notification panel according to screenshots.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.3 (tried 8.1.2 after this https://github.com/grafana/grafana/issues/39009#issuecomment-915948936 but same behavior)
- Data source type & version: Prometheus 2.29.1
- OS Grafana is installed on: optimized Linux OS on EKS 1.21
- User OS & Browser: Windows 10, Firefox developer 93
- Grafana plugins: none
- Others: setup on EKS done following this method https://grafana.com/docs/grafana/latest/installation/kubernetes/ with two envvar set additionnaly: `GF_FEATURE_TOGGLES_ENABLE=ngalert` + `GF_LOG_LEVEL=debug`

**Comment by loafoe** [comment]
Experiencing the same issue. I have a `Microsoft Teams` contact point configured. A number of `Cortex / Loki` rules are firing, but no notifications are triggered.

**Comment by damienleger** [comment]
I've just tried Grafana 8.1.4: same behavior.

**Comment by damienleger** [comment]
No improvement with Grafana 8.1.5. Any ideas?

**Comment by mucho-al-dente** [comment]
This is a bug in 8.1.3+. The solution is to set the alert interval to 30s or less

**Comment by damienleger** [comment]
@cianooooo 

hello, I've tried [this fix](https://github.com/grafana/grafana/issues/39295#issuecomment-924866713) soon after he appears and tweak my alerts interval, without success. I've just tried the version 8.2.0 (fix of the aforementioned is in 8.1.7), without success too.

![image](https://user-images.githubusercontent.com/16499022/136416030-834469d0-6308-43ed-aa41-e062ce58dae6.png)
![image](https://user-images.githubusercontent.com/16499022/136416161-26a4d7f6-5c38-4856-a63e-2bda38101683.png)

**Comment by mucho-al-dente** [comment]
@damienleger I believe you need to set the interval for the alert in the Alert Rules page. Edit the specific alert, and under section C 'Define Alert Conditions', set the interval there to 30s or less.

![image](https://user-images.githubusercontent.com/28299281/136760104-e374ddb1-0589-4018-b091-89b77bc95916.png)

**Comment by damienleger** [comment]
@cianooooo I see. But this field only appear on "alert managed via grafana". I don't have this value for "Cortex/Loki" alerts.
I found this value on prometheus and tweak to 15s (last screenshot). Still no notifications :slightly_frowning_face:

my alert rules panel:

![image](https://user-images.githubusercontent.com/16499022/136786111-fa10ba0f-3daf-437b-8770-160c61f9a201.png)
![image](https://user-images.githubusercontent.com/16499022/136786141-fa49515a-cadb-4d22-8b6c-51674bfd2433.png)

when click "new alert rule":
![image](https://user-images.githubusercontent.com/16499022/136786223-cc64cb27-fa53-480e-a138-83d1fa0575ba.png)
![image](https://user-images.githubusercontent.com/16499022/136786440-337d4403-4a7b-436f-a087-36d8557b8748.png)

tweak on prometheus:
![image](https://user-images.githubusercontent.com/16499022/136789477-22027d71-4b9e-44ad-948e-9f6e903373a1.png)

**Comment by damienleger** [comment]
I've created a "grafana managed alert" to check notification is working: it is! The problem is just with "Cortex / Loki" alerts.

![image](https://user-images.githubusercontent.com/16499022/136794348-99dfccef-7756-4c03-878d-208f60230c7b.png)
![image](https://user-images.githubusercontent.com/16499022/136794385-b385c28e-e0a4-4523-bfbe-ef28c4ba8166.png)

**Comment by grawert** [comment]
I can confirm this. Running Grafana `8.2.1`. Cortex/Loki alerts firing from Prometheus are listed in the alerts list. But no notification is send. Creating a Grafana alert rule with Prometheus data source is sending an email notification.

**Comment by wwalters12** [comment]
We are also experiencing this on Grafana `8.3.1` when deployed with the grafana/grafana helm chart on Kubernetes. 

We added our prometheus instances as datasources and can see the alerts tied to them in the alerting page, but creating notification policies for these alerts does not trigger a notification. We see no indication in the grafana pod logs that these alerts are being received by Grafana's alertmanager.  Like others have posted, creating a Grafana alert rule using the same prometheus data sources will trigger a notification.

There is also a post on the community forums describing the same issue: https://community.grafana.com/t/unified-alerts-granfana-8-notification-policies-based-on-prometheus-alerts/57316

**Comment by xafer337** [comment]
I'm having the same problem. The alerts firing in Prometheus are correctly shown in Grafana in the cortex/loki alert list, but they don't send any notifications. Creating the same alert in the grafana does trigger the notification.

Is it even possible to use the Grafana embedded alertmanager as an external alertmanager for Prometheus?

**Comment by mazzz1y** [comment]
the same issue

```
Grafana v8.3.3 (30bb7a93ca)
```

**Comment by MatthewScholefield** [comment]
I see alerts failing to go through when I have a webhook configured that fails. In my case it was the grafana image renderer which was failing because the new alerting system doesn't support it. Is this possibly everyone's issue here? Here's the logs I found related to this:
```
lvl=info msg="sending telegram notification" logger=alerting.notifier.telegram chat_id=-123456
lvl=eror msg="Failed to send webhook" logger=alerting.notifier.telegram error="Webhook response status 400 Bad Request" webhook="Telegram - Foo API Alerts"
lvl=info msg="level=error component=dispatcher msg=\"Notify for alerts failed\" num_alerts=4 err=\"Telegram - Foo API Alerts/telegram[0]: notify retry canceled due to unrecoverable error after 1 attempts: Webhook response status 400 Bad Request\"" logger=alertmanager org=1
```

After I removed `GF_RENDERING_CALLBACK_URL` (and `GF_RENDERING_SERVER_URL`) from my config, alerts seem to go through fine, but perhaps this is just conincidental and it is now just working after a restart.

**Comment by derN3rd** [comment]
> I see alerts failing to go through when I have a webhook configured that fails. In my case it was the grafana image renderer which was failing because the new alerting system doesn't support it. Is this possibly everyone's issue here? Here's the logs I found related to this:
> 
> ```
> lvl=info msg="sending telegram notification" logger=alerting.notifier.telegram chat_id=-123456
> lvl=eror msg="Failed to send webhook" logger=alerting.notifier.telegram error="Webhook response status 400 Bad Request" webhook="Telegram - Foo API Alerts"
> lvl=info msg="level=error component=dispatcher msg=\"Notify for alerts failed\" num_alerts=4 err=\"Telegram - Foo API Alerts/telegram[0]: notify retry canceled due to unrecoverable error after 1 attempts: Webhook response status 400 Bad Request\"" logger=alertmanager org=1
> ```
> 
> After I removed `GF_RENDERING_CALLBACK_URL` (and `GF_RENDERING_SERVER_URL`) from my config, alerts seem to go through fine, but perhaps this is just conincidental and it is now just working after a restart.

Had the same issue, but it was due to the new Alert grouping and telegrams message length limit, and not related to the image rendering.
If you have many alerts or just alerts with a large payload it might fail for Telegram as well

**Comment by alexsanderp** [comment]
I'm having the same problem. Grafana does not send Prometheus alerts.
Version: v8.3.3

Is there a possibility that this feature is only for viewing alerts in Grafana?

**Comment by alexsanderp** [comment]
> I see alerts failing to go through when I have a webhook configured that fails. In my case it was the grafana image renderer which was failing because the new alerting system doesn't support it. Is this possibly everyone's issue here? Here's the logs I found related to this:
> 
> ```
> lvl=info msg="sending telegram notification" logger=alerting.notifier.telegram chat_id=-123456
> lvl=eror msg="Failed to send webhook" logger=alerting.notifier.telegram error="Webhook response status 400 Bad Request" webhook="Telegram - Foo API Alerts"
> lvl=info msg="level=error component=dispatcher msg=\"Notify for alerts failed\" num_alerts=4 err=\"Telegram - Foo API Alerts/telegram[0]: notify retry canceled due to unrecoverable error after 1 attempts: Webhook response status 400 Bad Request\"" logger=alertmanager org=1
> ```
> 
> After I removed `GF_RENDERING_CALLBACK_URL` (and `GF_RENDERING_SERVER_URL`) from my config, alerts seem to go through fine, but perhaps this is just conincidental and it is now just working after a restart.

Would that be with the Prometheus alerts automatically imported? Or are they created by the interface?

**Comment by CCOLLOT** [comment]
Same issue here. Any comments from the Grafana Team?

**Comment by oleksandr-semak** [comment]
Just join to the issue we have the same issue

**Comment by kathy-lee** [comment]
Same issue with 8.3.4 for cortex/loki alerts, could not see slack notification message even they're in the "firing" state

**Comment by lfdominguez** [comment]
But these alerts are not sending to another alertmanager different than the internal alertmanager of grafana....grafana is configuring an internal alertmanager that can be different that one where prometheus is sending the alerts

**Comment by lfdominguez** [comment]
I think that the issue here can be a feature: `Allow Grafana modify external alertmanager configurations`

**Comment by oleksandr-semak** [comment]
@lfdominguez you right, also after some digging figured out that internal alertmanager which implemented in grafana not back-capable with prometheus alertmanager, but Grafana already has alertmanager datasource in `enable_alpha` so it allow see alerts and notifiers in Grafana Alerting Page and use Silences.
https://grafana.com/docs/grafana/latest/datasources/alertmanager/

**Comment by clglavan** [comment]
same behaviour on grafana 8.3.x - deployed with kube-prometheus-stack, adding contact points - testing said contact points work but the firing alerts never send a notification and in grafana logs I see no info relevant to alerting

**Comment by clglavan** [comment]
The solution that worked for me was to add the receiver to alertmanager config https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L160

**Comment by lfdominguez** [comment]
> The solution that worked for me was to add the receiver to alertmanager config https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L160

The thing is, for example, that grafana has support for Telegram notification, but if you need to modify directly the alertmanager config, you need an additional element, a webhook receiver that accept alertmanager notifications and send to telegram.... into grafana this is sooo easy, but not synced with stack alertmanager

**Comment by peter-dolkens** [comment]
I'm also having this issue - looking for a direct grafana -> slack integration

**Comment by SmallGaoX** [comment]
Is it possible that grafana just checks the alarms of prometheus and does not support sending notifications, although I am also looking forward to using grafana for unified alarming

**Comment by devminded** [comment]
As I understand it, Grafana unified alerting should have an built-in/internal alert manager named "Grafana" (it's just not visible in the alert panel when there is only one). It should act like a reconfigurable alertmanager and contact points and policies will be added to it; as can be seen under **alerts / admin.**

It does not seem to work. I even tried to re-add it under **alerts / admin / add external / as http://localhost:9093** which is found and register as healthy. Did not work. Tested with 8.3.3, 8.3.6, 8.4.0 OSS Docker container.

There is also nothing in the debug log about even attempting to send any notifications, but there are a lot of of the following:

```t=2022-02-18T15:59:48+0000 lvl=eror msg="unexpected content type from upstream. expected YAML, got text/html; charset=utf-8" logger=context userId=21 orgId=1 uname=daniel.sandberg error="unexpected content type from upstream. expected YAML, got text/html; charset=utf-8" remote_addr=10.10.keke.kaakaaaa
t=2022-02-18T15:59:52+0000 lvl=dbug msg="recording state cache metrics" logger=ngalert now=2022-02-18T15:59:52+0000
t=2022-02-18T15:59:57+0000 lvl=dbug msg="alert rules fetched" logger=ngalert count=0 disabled_orgs=[]
t=2022-02-18T16:00:07+0000 lvl=dbug msg="recording state cache metrics" logger=ngalert now=2022-02-18T16:00:07+0000
t=2022-02-18T16:00:07+0000 lvl=dbug msg="alert rules fetched" logger=ngalert count=0 disabled_orgs=[]
t=2022-02-18T16:00:07+0000 lvl=dbug msg="removing expired data encryption keys from cache..." logger=secrets
t=2022-02-18T16:00:07+0000 lvl=dbug msg="done removing expired data encryption keys from cache" logger=secrets
t=2022-02-18T16:00:08+0000 lvl=dbug msg="start of admin configuration sync" logger=ngalert
t=2022-02-18T16:00:08+0000 lvl=dbug msg="found admin configurations" logger=ngalert count=1
t=2022-02-18T16:00:08+0000 lvl=dbug msg="sender configuration is the same as the one running, no-op" logger=ngalert org=1 alertmanagers=[http://localhost:9093]
t=2022-02-18T16:00:08+0000 lvl=dbug msg="finish of admin configuration sync" logger=ngalert
t=2022-02-18T16:00:09+0000 lvl=dbug msg="synchronizing Alertmanagers for orgs" logger=ngalert.multiorg.alertmanager
t=2022-02-18T16:00:09+0000 lvl=dbug msg="neither config nor template have changed, skipping configuration sync." logger=alertmanager org=1
t=2022-02-18T16:00:09+0000 lvl=dbug msg="done synchronizing Alertmanagers for orgs" logger=ngalert.multiorg.alertmanager
t=2022-02-18T16:00:17+0000 lvl=dbug msg="alert rules fetched" logger=ngalert count=0 disabled_orgs=[]
```

One would expect the following line to be executed when notifying: https://github.com/grafana/grafana/blob/b2e6ca9393c3666956b7808f94d20a6dcc3586e0/pkg/services/ngalert/notifier/channels/alertmanager.go#L77

And these two if it fails to notify:
https://github.com/grafana/grafana/blob/b2e6ca9393c3666956b7808f94d20a6dcc3586e0/pkg/services/ngalert/notifier/channels/alertmanager.go#L97
https://github.com/grafana/grafana/blob/b2e6ca9393c3666956b7808f94d20a6dcc3586e0/pkg/services/ngalert/notifier/channels/alertmanager.go#L105

The actual embedded alertmanager seems to be configured here and should log lots of errors if it cannot properly start: https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/alertmanager.go

Notifications are actually never dispatched to the notifier/channel?

**EDIT:**
I'm not a Go programmer but I have kept digging. The actual scheduling of sending alerts is done in the schedular and in particular it loads the alerts here https://github.com/grafana/grafana/blob/7643ae6c5ef6ca143a0e049ebe3c1d60cbdb0f5a/pkg/services/ngalert/schedule/schedule.go#L380 and register a callback routine that does the actual notification. This is repeated each time the list of rules are updated, or at-least when their state is changed or by some evaluation timer? But it seems like it is always empty/zero `lvl=dbug msg="alert rules fetched" logger=ngalert count=0 disabled_orgs=[]` and we never get a new routine created? Or is the stuff in the alerts_rule table only "grafana managed alerts" and the prometheus alerts comes only exists in memory by polling the rules, alert APIs?

It feels like since we never get any "alert rule routine started" in the debug log even when updating the prometheus rules with a rule that always fire - something is off.

I might just be lost in the code.

**Comment by dvidben** [comment]
Continuing the analysis provided by @devminded  above, it looks like Grafana is only evaluating the rules defined in its own database:
https://github.com/grafana/grafana/blob/7643ae6c5ef6ca143a0e049ebe3c1d60cbdb0f5a/pkg/services/ngalert/store/alert_rule.go#L439

Is this the expected behavior? Are there any plans to extend this beyond?

**Comment by SmallGaoX** [comment]
I tried to create a new rule and found it in the alert_rule table in the grafana.db database as follows
```
[
    {
        "refId": "A",
        "queryType": "",
        "relativeTimeRange": {
            "from": 600,
            "to": 0
        },
        "datasourceUid": "PBFA97CFB590B2093",
        "model": {
            "exemplar": false,
            "expr": "100 - (avg by(instance) (rate(node_cpu_seconds_total{mode=\"idle\"}[2m])) * 100)",
            "hide": false,
            "interval": "",
            "intervalMs": 1000,
            "legendFormat": "",
            "maxDataPoints": 43200,
            "refId": "A"
        }
    },
    {
        "refId": "B",
        "queryType": "",
        "relativeTimeRange": {
            "from": 0,
            "to": 0
        },
        "datasourceUid": "-100",
        "model": {
            "conditions": [
                {
                    "evaluator": {
                        "params": [
                            80
                        ],
                        "type": "gt"
                    },
                    "operator": {
                        "type": "and"
                    },
                    "query": {
                        "params": [
                            "A"
                        ]
                    },
                    "reducer": {
                        "params": [],
                        "type": "last"
                    },
                    "type": "query"
                }
            ],
            "datasource": {
                "type": "__expr__",
                "uid": "-100"
            },
            "hide": false,
            "intervalMs": 1000,
            "maxDataPoints": 43200,
            "refId": "B",
            "type": "classic_conditions"
        }
    }
]
```
Can Prometheus' rules be converted to this type of JSON and stored there?

**Comment by armandgrillet** [comment]
@damienleger The contact points are set up for the internal AlertManager but there are no alert rules in this Alertmanager according to the first screenshot, this is why no notifications are being sent. 
![133401632-1b81208f-fa7d-4483-adc6-be735cd49552](https://user-images.githubusercontent.com/2117580/155294746-ba0c1f9f-36bd-4984-952e-f5db728e777f.png)

@peterholmberg is gonna open a PR where we clarify which Alertmanager is being used to set contact points even when there is only one configured, right now this is visible only if multiple Alertmanagers can have their contact points edited. It will look like this:

<img width="296" alt="PS 2022-02-23 à 10 44 53" src="https://user-images.githubusercontent.com/2117580/155294943-76de75ec-7fd3-43e4-a089-0ffbb3d17d9f.png">

**Reply by shohagrana64** [reply]
Any update on this?

**Comment by gotjosh** [comment]
@damienleger et al., 

# Grafana receiving alerts

Grafana **is not an alert receiver** is it an **alert generator**. This means that Grafana cannot receive alerts from anything else but its internal alert generator.

Receiving alerts from Prometheus (or anything else) is not supported at the time. Mainly because we didn't see a need for Grafana to receive alerts from the outside, we didn't want to encourage Grafana to be a direct replacement for the upstream Prometheus Alertmanager, and it would make alerting more complicated than what it already is. 

# Workflows

Our goal with unified alerting was to simplify and streamline the alerting workflow. This is very nuanced given Grafana supports 4 different types of systems that can speak the same language but offer different functionality:

Alert Generators
- Prometheus: Can generate alerts, **does not** have a way to configure rules via an API
- Cortex: Can generate alerts, **does** have a way to configure rules via an API

Alert Receivers
- Alertmanager: Can receive alerts, **does not** have a way to configure receivers, contact points, etc. via an API
- Cortex Alertmanger: Can receive alerts, **does** have a way to configure receivers, contact points, etc. via an API

While Grafana can generate ([and send](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/fundamentals/alertmanager/#alertmanager)) alerts, we decided to stop there. If anything, a recommendation for resiliency is to have your alert generator, and alert receiver be separate systems.

In the end, we support two workflows:

- Grafana generated alerts go to the internal Alertmanager or [an external Alertmanager](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/fundamentals/alertmanager/#alertmanager). You would be able to **see** everything within Grafana but have the option to **centralise alerts in the external Alertmanage**r.
- Prometheus generated alerts should go to an external Alertmanager. You would be able to **see** everything within Grafana, but **keep where your Grafana alerts and Prometheus alerts separate**.

# The proposal in this issue

The way I see it, the suggestion in this issue is to have a third option by having all of your alerts centralised in the Grafana Alertmanager. Although we would prefer if not as this opens the gate to even more complexity in alerting with enough support, we could consider the idea. 

I'm turning this issue into a feature request to allow others to chime in and express their opinions but, more importantly, make this feature request more discoverable.

**Reply by espendk** [reply]
The reluctance to add this feature clashes with Grafana Cloud being positioned as an all-in-one ops solution with monitoring, alerting, alert mgmt, on-call mgmt, and incident mgmt. 

Providing all these capabilities **but** only partial alert mgmt in Grafana Cloud seems quite an odd choice by Grafana Labs..

**Reply by gotjosh** [reply]
Sorry @espendk, I don't think you're correct. Grafana cloud comes bundled with an Alertmanager **instance** that lives outside of Grafana. It is important to note that there's a subtle but fundamental distinction between the two in that Grafana Alerting is not **JUST** the Alertmanger.

**Reply by belm0** [reply]
> This means that Grafana cannot receive alerts from anything else but its internal alert generator.

In that case, calling the 8.x enhancements "unified alerting" was a little misleading.  If it's necessary to keep parallel configuration of routing and notification endpoints in both Grafana and Prometheus alertmanager, that isn't unified.

> Although we would prefer if not as this opens the gate to even more complexity in alerting with enough support, we could consider the idea.

Since the Grafana alert rules page can enumerate alert status across all the sources (both internal and external), it doesn't seem that far off to send them all to the same notification handling.

**Comment by devminded** [comment]
This clears it up a lot. I think just having these things clarified in the documentation.

> Alert notifications from Grafana only applies to "Grafana managed alerts". You would still be able to see everything within Grafana, including alerts from Prometheus, but the Prometheus alerts will *not* be propagated by Grafana itself.
> Prometheus generated alerts should go to an external Alertmanager, configured in Prometheus itself, not within Grafana.
> Grafana managed alerts could either be propagated to an external alertmanager (contact point and routing policy) or added as external alertmanagers under "alerts / admin" that will receive all Grafana managed alerts.

**Reply by Mark90** [reply]
Yeah. I'm still a novice at Grafana/Prometheus, and despite reading all of the documentation regarding alerts this distinction did not become clear to me. Though perhaps the information is there and I just failed to recognize it.

I can see why adding the alert receiving logic into Grafana would introduce too much complexity, and it not being the right application for it. But then maybe Grafana could at least show an "(info)" icon in the Alerting window near the "Mimir / Cortex / Loki" section which clarifies that one cannot connect these to any notification policies. That would save beginners like me some debugging time. :)

**Comment by SmallGaoX** [comment]
Is the proposal likely to be adopted

**Comment by tmyhu** [comment]
This would be extremely useful in our environment. We have a large number of segregated, disconnected networks, each with a prometheus instance that has hundreds of predefined alerts (OpenShift).
We can only connect into those environments to pull data but we are not allowed to connect out to send alerts. We can now see all alerts in a central Grafana but it would be a huge improvement if they would trigger notifications from that central Grafana which is in a network segment allowed to connect to the outside world for alerting.

**Comment by R-Studio** [comment]
I hope this feature request will come! 😃

**Comment by legolego621** [comment]
This is really important to many of us

**Comment by hpl002** [comment]
As mentioned [here](https://github.com/grafana/grafana/discussions/45773#discussioncomment-2235419) it would be useful to be able to convert prometheus alerts into the provisioned alerts schema.
I'm sure a lot of us would happily jump over to grafana alerts if this was made available.

On a related note;
Are there any community collections of grafana alerts that could be used as a starting point. Rewriting hundreds of prometheus is too time consuming..

**Comment by danielcb** [comment]
Just for clarification:

1. There is no way to use oncall or Grafana alert notifications in general without using grafana alerts.
2. There is currently no tooling to easily convert a collection of prometheus alerts into grafana alerts.

Are both those statements correct?

**Reply by gotjosh** [reply]
>There is no way to use oncall or Grafana alert notifications in general without using grafana alerts.

I'm not sure I follow this question - Oncall as a product is completely separate from Grafana so naturally it supports receiving alerts from anything that implements its APIs.

> There is currently no tooling to easily convert a collection of prometheus alerts into grafana alerts.
There's no currently no tooling that does this and this is not something we'd recommend. If your data is Prometheus-based, there should be no need use Grafana Alerting. 

It'll be more reliable, faster and less resource intensive to evaluate your alerts within Prometheus.

**Comment by yp05327** [comment]
![image](https://github.com/grafana/grafana/assets/18380374/f03cdb37-7227-4e58-be52-678253cca906)
The official plugin `Prometheus` is marked as `Alerting supported`, but you said it is not supported. 😕 
I think you should convert this incorrect tag into `Alerting not supported`, to avoid other users wasting time testing this non-existing feature.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/73447).
